src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDeclarationFix.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.PXFieldAttributes;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Acuminator.Analyzers.StaticAnalysis.DacKeyFieldDeclaration
+{
+	[Shared]
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class KeyFieldDeclarationFix : CodeFixProvider
+	{
+		private const string IsKey = "IsKey";
+		private const string True = "true";
+			
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(Descriptors.PX1055_DacKeyFieldBound.Id);
+
+		public override  Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			return Task.Run(() =>
+			{
+				var diagnostic = context.Diagnostics.FirstOrDefault(d => d.Id == Descriptors.PX1055_DacKeyFieldBound.Id);
+
+				if (diagnostic == null || context.CancellationToken.IsCancellationRequested)
```

Comment:
We have a code style rule to use `ThrowIfCancellationRequested` instead of `IsCancellationRequested`